File: core/src/galenscovell/hinterstar/ui/components/gamescreen/hud/EventPanel.scala
package galenscovell.hinterstar.ui.components.gamescreen.hud

import com.badlogic.gdx.scenes.scene2d.ui._
import com.badlogic.gdx.utils.Align
import galenscovell.hinterstar.processing.Event
import galenscovell.hinterstar.ui.components.gamescreen.stages.HudStage
import galenscovell.hinterstar.util._


class EventPanel(stage: HudStage, eventContainer: Event) extends Table {
  private val hudStage: HudStage = stage
  private val container: Event = eventContainer

  construct()


  def construct(): Unit = {
    this.setFillParent(true)
    val mainTable: Table = new Table
    mainTable.setBackground(Resources.npTest4)

    val titleTable: Table = createTitleTable(container.name)

    val centerTable: Table = new Table
    val descriptionTable: Table = createDescriptionTable(container.description)
    val optionTable: Table = createOptionTable

    centerTable.add(descriptionTable).height(100).expand.fill.top
    centerTable.row
    centerTable.add(optionTable).width(680).height(250).expand.fill

    mainTable.add(titleTable).height(50).expand.fill.top
    mainTable.row
    mainTable.add(centerTable).height(350).expand.fill.top

    this.add(mainTable).width(680).height(400).expand.fill.center
  }

  private def createTitleTable(eventTitle: String): Table = {
    val titleTable: Table = new Table
    titleTable.setBackground(Resources.npTest1)
    val titleLabel: Label = new Label(eventTitle, Resources.labelTitleStyle)
    titleLabel.setAlignment(Align.center, Align.center)
    titleTable.add(titleLabel)
    titleTable
  }

  private def createDescriptionTable(eventDescription: String): Table = {
    val descriptionTable: Table = new Table
    val descriptionLabel: Label = new Label(eventDescription, Resources.labelDetailStyle)
    descriptionLabel.setWrap(true)
    descriptionLabel.setAlignment(Align.center, Align.left)
    descriptionTable.add(descriptionLabel).width(680)
    descriptionTable
  }

  private def createOptionTable: Table = {
    val optionTable: Table = new Table
    optionTable.setBackground(Resources.npTest1)

    for (choice <- container.choices) {
      val choiceText: String = choice.get("choice-text") match {
        case Some(a) => s"$a"
        case None    => "ERROR: Choice text not found"
      }
      val choiceButton: TextButton = new TextButton(choiceText, Resources.buttonEventStyle)
      choiceButton.getLabel.setAlignment(Align.left, Align.left)
      choiceButton.getLabel.setWrap(true)
      choiceButton.getLabelCell.padLeft(20).padRight(20)
      optionTable.add(choiceButton).width(680).height(50).expand.fill.top.padTop(2).padBottom(2)
      optionTable.row
    }

    optionTable
  }
}

File: core/src/galenscovell/hinterstar/ui/components/gamescreen/hud/EventPanel.kt
package galenscovell.hinterstar.ui.components.gamescreen.hud

import com.badlogic.gdx.scenes.scene2d.ui.*
import com.badlogic.gdx.utils.Align
import galenscovell.hinterstar.processing.Event
import galenscovell.hinterstar.ui.components.gamescreen.stages.InterfaceStage
import galenscovell.hinterstar.util.Resources


class EventPanel(private val interfaceStage: InterfaceStage, private val eventContainer: Event) : Table() {

    init {
        this.setFillParent(true)
        val mainTable: Table = Table()
        mainTable.background = Resources.npTest4

        val titleTable: Table = createTitleTable(eventContainer.name)

        val centerTable: Table = Table()
        val descriptionTable: Table = createDescriptionTable(eventContainer.description)
        val optionTable: Table = createOptionTable()

        centerTable.add(descriptionTable).height(100f).expand().fill().top()
        centerTable.row()
        centerTable.add(optionTable).width(680f).height(250f).expand().fill()

        mainTable.add(titleTable).height(50f).expand().fill().top()
        mainTable.row()
        mainTable.add(centerTable).height(350f).expand().fill().top()

        this.add(mainTable).width(680f).height(400f).expand().fill().center()
    }



    private fun createTitleTable(eventTitle: String): Table {
        val titleTable: Table = Table()
        titleTable.background = Resources.npTest1
        val titleLabel: Label = Label(eventTitle, Resources.labelXLargeStyle)
        titleLabel.setAlignment(Align.center, Align.center)
        titleTable.add(titleLabel)
        return titleTable
    }

    private fun createDescriptionTable(eventDescription: String): Table {
        val descriptionTable: Table = Table()
        val descriptionLabel: Label = Label(eventDescription, Resources.labelSmallStyle)
        descriptionLabel.setWrap(true)
        descriptionLabel.setAlignment(Align.center, Align.left)
        descriptionTable.add(descriptionLabel).width(680f)
        return descriptionTable
    }

    private fun createOptionTable(): Table {
        val optionTable: Table = Table()
        optionTable.background = Resources.npTest1

        for (choice in eventContainer.choices) {
            val choiceText: String = choice["choice-text"].toString()
            val choiceButton: TextButton = TextButton(choiceText, Resources.buttonEventStyle)
            choiceButton.label.setAlignment(Align.left, Align.left)
            choiceButton.label.setWrap(true)
            choiceButton.labelCell.padLeft(20f).padRight(20f)
            optionTable.add(choiceButton).width(680f).height(50f).expand().fill().top().padTop(2f).padBottom(2f)
            optionTable.row()
        }

        return optionTable
    }
}
